gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -752,25 +745,16 @@ private SeekableByteChannel open(
       throw createFileNotFoundException(
           resourceId.getBucketName(), resourceId.getObjectName(), /* cause= */ null);
     }
+
     if (storageOptions.isGrpcEnabled()) {
-      return itemInfo == null
-          ? new GoogleCloudStorageGrpcReadChannel(
-              storageStubProvider,
-              storage,
-              resourceId,
-              watchdog,
-              metricsRecorder,
-              readOptions,
-              BackOffFactory.DEFAULT,
-              this.storageOptions)
-          : new GoogleCloudStorageGrpcReadChannel(
-              storageStubProvider,
-              itemInfo,
-              watchdog,
-              metricsRecorder,
-              readOptions,
-              BackOffFactory.DEFAULT,
-              this.storageOptions);
+      return new GoogleCloudStorageGrpcReadChannel(
+          storageStubProvider,
+          itemInfo == null ? getItemInfo(resourceId) : itemInfo,
```

Comment:
This PR is just cleaning up the metadata call but not adding support for failFast logic for gRPC. Is my understanding correct?

Reply:
Yes, I think that gRPC channel as it written right now always requires metadata call/info - this can be improved as follow up, if possible.